<strong>SAE 1.04 Création d'une base de données</strong>

<strong>SAE BDD de SARAVANAN Tamij et SADI OUFELLA Kenza</strong>

<strong>Phase 1: Compréhension et Analyse </strong>

<p>Cardinalité :891 tuples
Degrés :12

Attribut PassengerId= Numéro de passager va de 1 à 891

Attribut Survived {0,1}(0 = mort / 1 = vivant)
Il y a eu 549 morts et 342 survivant

Attribut PClasse {1,2,3}I
il y a eu 216 tickets de 1ere classe vendue, 184 pour la 2eme classe et 491 tickets pour la 3eme classe

Attribut Name = Nom et Prénom

Attribut sex {male, female}
65% étaient des hommes contre 35% des femmes

Attribut Age= age des passager (entier positif or null)

Attribut SibSp= nb de  frère et soeur/ conjoint à bord (entier positif)

Attribut Parch = nb de parent/ enfant à bord (entier)

Attribut Ticket = Numéro de ticket= différent les une et les autres

Attribut Fare = tarif des passager (entier positif)

Attribut Cabin = Numéro de Cabine, beaucoup de case sont vides, donc il faut considérer qu’il sont null 

Attribut Embarked  {Q, S, C}(Q = Queenstone / S = Southampton / C = Cherbourg) = Le port où l’embarquement à été fait
72% des passagers ont embarqué à Southampton, 19% ont embarqué àCherbourg et 9% ont embarqué à Queenstone
</p>

<strong>Phase 2 : création et alimentation des Tables</strong>

In [2]:
%reload_ext sql
print("c'est fait")

c'est fait


In [3]:
%sql postgresql://postgres:0000@localhost/postgres
print("connecter")


connecter


In [5]:
%%sql
drop view vivantclass1;
drop view vivantclass2;
drop view vivantclass3;
drop view mortclass1;
drop view mortclass2;
drop view mortclass3;
drop view vivantmale;
drop view vivantfemale;
drop view mortmale;
drop view mortfemale;
drop table Survivants;
drop table Tickets;
drop table Passagers;
drop table titanic;

 * postgresql://postgres:***@localhost/postgres
Done.
Done.
Done.
Done.


[]

In [6]:
%%sql

create table titanic(
PassengerId integer,
Survived integer,
Pclass integer,
Name varchar(100),
Sex varchar(10),
Age float,
SibSp integer,
Parch integer,
numTicket varchar(20),
Fare float,
Cabin varchar(20),
Embarked varchar(1));

create table Passagers(
PassengerId integer primary key,
Name varchar(100),
Sex varchar(10),
Age float,
SibSp integer,
Parch integer);

create table Tickets(
Pclass integer,
Name varchar(100) primary key,
numTicket varchar(20),
Fare float,
Age float,
Cabin varchar(20),
Embarked varchar(1));

create table Survivants(
PassengerId integer,
Survived integer,
Name varchar (100),
primary key (PassengerId, Name),
foreign key (PassengerId) references Passagers(PassengerId),
foreign key (Name) references Tickets(Name));

 * postgresql://postgres:***@localhost/postgres
Done.
Done.
Done.
Done.


[]

In [7]:
%%sql
copy titanic from '/home/saravanan/Bureau/titanic_train.csv' delimiter ',' CSV header;


 * postgresql://postgres:***@localhost/postgres
891 rows affected.


[]

In [8]:
%%sql
insert into Passagers (PassengerId, Name, Sex, Age, SibSp, Parch) select PassengerId, Name, Sex, Age, SibSp, Parch from titanic;

 * postgresql://postgres:***@localhost/postgres
891 rows affected.


[]

In [9]:
%%sql
insert into Tickets (Pclass, Name, numTicket, Fare,Age, Cabin, Embarked) select Pclass, Name, numTicket, Fare,Age,Cabin, Embarked from titanic;

 * postgresql://postgres:***@localhost/postgres
891 rows affected.


[]

In [10]:
%%sql
insert into Survivants (PassengerId, Survived, Name) select PassengerID, Survived, Name from titanic;

 * postgresql://postgres:***@localhost/postgres
891 rows affected.


[]

In [11]:
result = %sql select * from titanic;
result1= %sql select * from Passagers;
result2= %sql select * from Tickets;
result3= %sql select * from Survivants;

 * postgresql://postgres:***@localhost/postgres
891 rows affected.
 * postgresql://postgres:***@localhost/postgres
891 rows affected.
 * postgresql://postgres:***@localhost/postgres
891 rows affected.
 * postgresql://postgres:***@localhost/postgres
891 rows affected.


In [12]:
dataframe = result.DataFrame()
dataframe

passengerid  survived  pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  name     sex   age  sibsp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     parch         numticket     fare cabin embarked  
0        0         A/5 21171   7.2500  None        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250  None        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500  None        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000  None        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500  None        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500  None        Q  

[891 rows x 12 columns]

In [13]:
dataframe = result1.DataFrame()
dataframe

passengerid                                               name     sex  \
0              1                            Braund, Mr. Owen Harris    male   
1              2  Cumings, Mrs. John Bradley (Florence Briggs Th...  female   
2              3                             Heikkinen, Miss. Laina  female   
3              4       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   
4              5                           Allen, Mr. William Henry    male   
..           ...                                                ...     ...   
886          887                              Montvila, Rev. Juozas    male   
887          888                       Graham, Miss. Margaret Edith  female   
888          889           Johnston, Miss. Catherine Helen "Carrie"  female   
889          890                              Behr, Mr. Karl Howell    male   
890          891                                Dooley, Mr. Patrick    male   

      age  sibsp  parch  
0    22.0      1      0  
1    38.0      1      0  
2    26.0      0      0  
3    35.0      1      0  
4    35.0      0      0  
..    ...    ...    ...  
886  27.0      0      0  
887  19.0      0      0  
888   NaN      1      2  
889  26.0      0      0  
890  32.0      0      0  

[891 rows x 6 columns]

In [14]:
dataframe = result2.DataFrame()
dataframe

pclass                                               name  \
0         3                            Braund, Mr. Owen Harris   
1         1  Cumings, Mrs. John Bradley (Florence Briggs Th...   
2         3                             Heikkinen, Miss. Laina   
3         1       Futrelle, Mrs. Jacques Heath (Lily May Peel)   
4         3                           Allen, Mr. William Henry   
..      ...                                                ...   
886       2                              Montvila, Rev. Juozas   
887       1                       Graham, Miss. Margaret Edith   
888       3           Johnston, Miss. Catherine Helen "Carrie"   
889       1                              Behr, Mr. Karl Howell   
890       3                                Dooley, Mr. Patrick   

            numticket     fare   age cabin embarked  
0           A/5 21171   7.2500  22.0  None        S  
1            PC 17599  71.2833  38.0   C85        C  
2    STON/O2. 3101282   7.9250  26.0  None        S  
3              113803  53.1000  35.0  C123        S  
4              373450   8.0500  35.0  None        S  
..                ...      ...   ...   ...      ...  
886            211536  13.0000  27.0  None        S  
887            112053  30.0000  19.0   B42        S  
888        W./C. 6607  23.4500   NaN  None        S  
889            111369  30.0000  26.0  C148        C  
890            370376   7.7500  32.0  None        Q  

[891 rows x 7 columns]

In [15]:
dataframe = result3.DataFrame()
dataframe

passengerid  survived                                               name
0              1         0                            Braund, Mr. Owen Harris
1              2         1  Cumings, Mrs. John Bradley (Florence Briggs Th...
2              3         1                             Heikkinen, Miss. Laina
3              4         1       Futrelle, Mrs. Jacques Heath (Lily May Peel)
4              5         0                           Allen, Mr. William Henry
..           ...       ...                                                ...
886          887         0                              Montvila, Rev. Juozas
887          888         1                       Graham, Miss. Margaret Edith
888          889         0           Johnston, Miss. Catherine Helen "Carrie"
889          890         1                              Behr, Mr. Karl Howell
890          891         0                                Dooley, Mr. Patrick

[891 rows x 3 columns]

<strong>Rapport:</strong>
<p>Nous avons commencé à créer la table titanic qui regroupe tous les attributs qui sont présents dans le fichier titanic_train.csv.
Cette table va nous permettre d'alimenter les autres tables qui sont : passager, ticket, famille grâce à la commande insert into .... (...,les attributs) select .... from titanic;
<br/>
</p>
<p>
Description des tables:
<br/>
- table Passagers:
les attributs sont : PassengersId integer (identification du passager),Name varchar(100),Sex varchar(10),Age float, Sibsp integer: le nombre de frères et soeurs ou conjoint de passagers, Parch integer :le nombre de parents ou d'enfants du passager.
cette table sert donc à récupérer les informations sur les passagers du bateau,nous avons choisi de mettre en clé primaire PassengersId 
<br/>
- Table Tickets :
Les attributs sont : Pclass integer,Name varchar(100),Age float,numTicket varchar(20),Fare float,Cabin varchar(20),Embarked varchar(1).
cette table permet de récupérer les informations par rapport au ticket donc le numéro de ticket,la classe,le lieu d'embarquement,le prix,Age pour voir si le prix du ticket dépend de l'âge et le numéro de cabine
nous avons décidé de mettre Name en clé primaire
<br/>
- Table Survivants:
Les attributs sont: PassengersId integer,Survived integer,Name varchar (100), 
cette tables permet de nous fournir les informations sur les survivants ,en fonction de son identifiant PassengersId et Name qui sont les clés primaire,et nous avons aussi des clés étrangères:
foreign key (PassengersId) references Passagers (PassengersId)
foreign key (Name) references Tickets(Name) donc il faut crée la table survivants à la fin 

</p>

<strong>Phase 3: Requêtes SQL </strong>

<p>Exemple de requête</p>

In [16]:
result4 = %sql select * from tickets where pclass=3;
print(result4[2])

 * postgresql://postgres:***@localhost/postgres
491 rows affected.
(3, 'Allen, Mr. William Henry', '373450', 8.05, 35.0, None, 'S')


In [17]:
result5= %sql select name from Passagers natural join Survivants where survived=1;
print(result5)

 * postgresql://postgres:***@localhost/postgres
342 rows affected.
+------------------------------------------------------------------------------------+
|                                        name                                        |
+------------------------------------------------------------------------------------+
|                Cumings, Mrs. John Bradley (Florence Briggs Thayer)                 |
|                               Heikkinen, Miss. Laina                               |
|                    Futrelle, Mrs. Jacques Heath (Lily May Peel)                    |
|                 Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)                  |
|                        Nasser, Mrs. Nicholas (Adele Achem)                         |
|                          Sandstrom, Miss. Marguerite Rut                           |
|                              Bonnell, Miss. Elizabeth                              |
|                          Hewlett, Mrs. (Mary D Kingcome)     

<strong>Réponse aux requêtes demandés</strong>

<em>Questions 1 : Combien de classes de passagers différentes y avait-il à bord du Titanic ?</em>

In [19]:
%%sql
select distinct pclass from Tickets group by pclass;

 * postgresql://postgres:***@localhost/postgres
3 rows affected.


[(3,), (2,), (1,)]

In [20]:
%%sql
select count(distinct pclass) from tickets;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(3,)]

In [43]:
import pandas
from sqlalchemy import create_engine
import plotly.express as px
engine = create_engine("postgresql://postgres:0000@localhost/postgres")
df1 = pandas.read_sql("select distinct pclass from tickets;", engine)
fig = px.bar(df1, y="pclass",color="pclass")
fig.show()

<p>Donc il y a au total 3 classes de passagers différentes à bord du Titanic</p>

<em>Questions 2 : Combien de passagers y avait-il dans chaque classe ?</em>


In [21]:
%%sql
select count(passengerid), pclass from passagers left outer join tickets on passagers.name=tickets.name group by pclass order by pclass;

 * postgresql://postgres:***@localhost/postgres
3 rows affected.


[(216, 1), (184, 2), (491, 3)]

<p> Donc il y a 216 passagers dans la classe 1, 184 passagers dans la classe 2, 491 passagers dans la classe 3.
</p>

In [49]:
import pandas
from sqlalchemy import create_engine
import plotly.express as px

engine = create_engine("postgresql://postgres:0000@localhost/postgres")
df1 = pandas.read_sql("select count(passengerid), pclass from passagers left outer join tickets on passagers.name=tickets.name group by pclass order by pclass;", engine)
fig = px.bar(df1, x="pclass",y="count",color="count")
fig.show()

<em>Questions 3 : Combien de femmes et d'hommes y avait-il dans chaque classe ?</em>

In [22]:
%%sql
select pclass, count(sex), sex from passagers left outer join tickets on passagers.name=tickets.name group by sex, pclass order by pclass;

 * postgresql://postgres:***@localhost/postgres
6 rows affected.


[(1, 94, 'female'),
 (1, 122, 'male'),
 (2, 108, 'male'),
 (2, 76, 'female'),
 (3, 144, 'female'),
 (3, 347, 'male')]

il y a 122 hommes et 94 femmes dans la classe 1

il y a 108 hommes et 76 femmes dans la classe 2

il y a 347 hommes et 144 femmes dans la classe 3

In [50]:
import pandas
from sqlalchemy import create_engine
import plotly.express as px

engine = create_engine("postgresql://postgres:0000@localhost/postgres")
df1 = pandas.read_sql("select pclass, count(sex),sex from passagers left outer join tickets on passagers.name=tickets.name group by sex, pclass order by pclass ;", engine)
fig = px.bar(df1, x="pclass",y="count", color="sex")
fig.show()

<em>Questions 4 : Comptez le nombre et le pourcentage de survivants et de passagers morts.</em>

In [69]:
%%sql
select count(survived)as nombre, count(survived)*100 / (select count(*) from survivants) as pourcentage from survivants where survived=1;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(342, 38)]

In [ ]:
Donc il y a 342 survivants soit 38% des passagers

In [70]:
%%sql
select count(survived)as nombre, count(survived)*100 / (select count(*) from survivants) as pourcentage from survivants where survived=0;

 * postgresql://postgres:***@localhost/postgres
1 rows affected.


[(549, 61)]

In [47]:
import pandas
from sqlalchemy import create_engine
import plotly.express as px

engine = create_engine("postgresql://postgres:0000@localhost/postgres")
df1 = pandas.read_sql("select count(survived),survived from survivants group by survived;", engine)
fig = px.bar(df1, x="survived", y = "count",color="count")
fig.show()

<em>Questions 5 : Visualiser la répartition des passagers survivants et morts par classe</em>

In [27]:
%%sql
select pclass, count(survived),survived from tickets left outer join survivants on survivants.name=tickets.name group by pclass,survived order by pclass;

 * postgresql://postgres:***@localhost/postgres
6 rows affected.


[(1, 80, 0), (1, 136, 1), (2, 87, 1), (2, 97, 0), (3, 372, 0), (3, 119, 1)]

In [71]:
%%sql 
create view vivantclass1 as select PassengerId,Name,survived from survivants natural join tickets where survived=1 and pclass=1 order by passengerid;
create view vivantclass2 as select PassengerId,Name,survived from survivants natural join tickets where survived=1 and pclass=2 order by passengerid;
create view vivantclass3 as select PassengerId,Name,survived from survivants natural join tickets where survived=1 and pclass=3 order by passengerid;

create view mortclass1 as select PassengerId,name,survived from survivants natural join tickets where survived=0 and pclass=1 order by passengerid;
create view mortclass2 as select PassengerId,name,survived from survivants natural join tickets where survived=0 and pclass=2 order by passengerid;
create view mortclass3 as select PassengerId,name,survived from survivants natural join tickets where survived=0 and pclass=3 order by passengerid;

 * postgresql://postgres:***@localhost/postgres
Done.
Done.
Done.
Done.
Done.
Done.


[]

In [72]:
%%sql
select * from vivantclass1;
select * from vivantclass2;
select * from vivantclass3;
select * from mortclass1;
select * from mortclass2;
select * from mortclass3;

 * postgresql://postgres:***@localhost/postgres
136 rows affected.
87 rows affected.
119 rows affected.
80 rows affected.
97 rows affected.
372 rows affected.


[(1, 'Braund, Mr. Owen Harris', 0),
 (5, 'Allen, Mr. William Henry', 0),
 (6, 'Moran, Mr. James', 0),
 (8, 'Palsson, Master. Gosta Leonard', 0),
 (13, 'Saundercock, Mr. William Henry', 0),
 (14, 'Andersson, Mr. Anders Johan', 0),
 (15, 'Vestrom, Miss. Hulda Amanda Adolfina', 0),
 (17, 'Rice, Master. Eugene', 0),
 (19, 'Vander Planke, Mrs. Julius (Emelia Maria Vandemoortele)', 0),
 (25, 'Palsson, Miss. Torborg Danira', 0),
 (27, 'Emir, Mr. Farred Chehab', 0),
 (30, 'Todoroff, Mr. Lalio', 0),
 (38, 'Cann, Mr. Ernest Charles', 0),
 (39, 'Vander Planke, Miss. Augusta Maria', 0),
 (41, 'Ahlin, Mrs. Johan (Johanna Persdotter Larsson)', 0),
 (43, 'Kraeff, Mr. Theodor', 0),
 (46, 'Rogers, Mr. William John', 0),
 (47, 'Lennon, Mr. Denis', 0),
 (49, 'Samaan, Mr. Youssef', 0),
 (50, 'Arnold-Franchi, Mrs. Josef (Josefine Franchi)', 0),
 (51, 'Panula, Master. Juha Niilo', 0),
 (52, 'Nosworthy, Mr. Richard Cater', 0),
 (58, 'Novel, Mr. Mansouer', 0),
 (60, 'Goodwin, Master. William Frederick', 0),
 (61, 'Sirayanian, Mr. Orsen', 0),
 (64, 'Skoog, Master. Harald', 0),
 (68, 'Crease, Mr. Ernest James', 0),
 (70, 'Kink, Mr. Vincenz', 0),
 (72, 'Goodwin, Miss. Lillian Amy', 0),
 (74, 'Chronopoulos, Mr. Apostolos', 0),
 (76, 'Moen, Mr. Sigurd Hansen', 0),
 (77, 'Staneff, Mr. Ivan', 0),
 (78, 'Moutal, Mr. Rahamin Haim', 0),
 (81, 'Waelens, Mr. Achille', 0),
 (87, 'Ford, Mr. William Neal', 0),
 (88, 'Slocovski, Mr. Selman Francis', 0),
 (90, 'Celotti, Mr. Francesco', 0),
 (91, 'Christmann, Mr. Emil', 0),
 (92, 'Andreasson, Mr. Paul Edvin', 0),
 (94, 'Dean, Mr. Bertram Frank', 0),
 (95, 'Coxon, Mr. Daniel', 0),
 (96, 'Shorney, Mr. Charles Joseph', 0),
 (101, 'Petranec, Miss. Matilda', 0),
 (102, 'Petroff, Mr. Pastcho ("Pentcho")', 0),
 (104, 'Johansson, Mr. Gustaf Joel', 0),
 (105, 'Gustafsson, Mr. Anders Vilhelm', 0),
 (106, 'Mionoff, Mr. Stoytcho', 0),
 (109, 'Rekic, Mr. Tido', 0),
 (112, 'Zabour, Miss. Hileni', 0),
 (113, 'Barton, Mr. David John', 0),
 (114, 'Jussila, Miss. Katriina', 0),
 (115, 'Attalah, Miss. Malake', 0),
 (116, 'Pekoniemi, Mr. Edvard', 0),
 (117, 'Connors, Mr. Patrick', 0),
 (120, 'Andersson, Miss. Ellis Anna Maria', 0),
 (122, 'Moore, Mr. Leonard Charles', 0),
 (127, 'McMahon, Mr. Martin', 0),
 (130, 'Ekstrom, Mr. Johan', 0),
 (131, 'Drazenoic, Mr. Jozef', 0),
 (132, 'Coelho, Mr. Domingos Fernandeo', 0),
 (133, 'Robins, Mrs. Alexander A (Grace Charity Laury)', 0),
 (139, 'Osen, Mr. Olaf Elon', 0),
 (141, 'Boulos, Mrs. Joseph (Sultana)', 0),
 (144, 'Burke, Mr. Jeremiah', 0),
 (148, 'Ford, Miss. Robina Maggie "Ruby"', 0),
 (153, 'Meo, Mr. Alfonzo', 0),
 (154, 'van Billiard, Mr. Austin Blyler', 0),
 (155, 'Olsen, Mr. Ole Martin', 0),
 (158, 'Corn, Mr. Harry', 0),
 (159, 'Smiljanic, Mr. Mile', 0),
 (160, 'Sage, Master. Thomas Henry', 0),
 (161, 'Cribb, Mr. John Hatfield', 0),
 (163, 'Bengtsson, Mr. John Viktor', 0),
 (164, 'Calic, Mr. Jovo', 0),
 (165, 'Panula, Master. Eino Viljami', 0),
 (168, 'Skoog, Mrs. William (Anna Bernhardina Karlsson)', 0),
 (170, 'Ling, Mr. Lee', 0),
 (172, 'Rice, Master. Arthur', 0),
 (174, 'Sivola, Mr. Antti Wilhelm', 0),
 (176, 'Klasen, Mr. Klas Albin', 0),
 (177, 'Lefebre, Master. Henry Forbes', 0),
 (180, 'Leonard, Mr. Lionel', 0),
 (181, 'Sage, Miss. Constance Gladys', 0),
 (183, 'Asplund, Master. Clarence Gustaf Hugo', 0),
 (189, 'Bourke, Mr. John', 0),
 (190, 'Turcin, Mr. Stjepan', 0),
 (197, 'Mernagh, Mr. Robert', 0),
 (198, 'Olsen, Mr. Karl Siegwart Andreas', 0),
 (201, 'Vande Walle, Mr. Nestor Cyriel', 0),
 (202, 'Sage, Mr. Frederick', 0),
 (203, 'Johanson, Mr. Jakob Alfred', 0),
 (204, 'Youseff, Mr. Gerious', 0),
 (206, 'Strom, Miss. Telma Matilda', 0),
 (207, 'Backstrom, Mr. Karl Alfred', 0),
 (211, 'Ali, Mr. Ahmed', 0),
 (213, 'Perkin, Mr. John Henry', 0),
 (215, 'Kiernan, Mr. Philip', 0),
 (223, 'Green, Mr. George Henry', 0),
 (224, 'Nenkoff, Mr. Christo', 0),
 (226, 'Berglund, Mr. Karl Ivar Sven', 0),
 (228, 'Lovell, Mr. John Hall ("Henry")', 0),
 (230, 'Lefebre, Miss. Mathilde', 0),
 (23

In [51]:
import pandas
from sqlalchemy import create_engine
import plotly.express as px

engine = create_engine("postgresql://postgres:0000@localhost/postgres")
df1 = pandas.read_sql("select pclass, count(survived),survived from tickets left outer join survivants on survivants.name=tickets.name group by pclass,survived order by survived ;", engine)
fig = px.bar(df1, x="pclass", y = "count", color = "survived")
fig.show()

<em>Questions 6 : Visualiser la répartition des passagers survivants et des passagers décéés selon le sexe.</em>

In [29]:
%%sql
select sex, count(survived),survived from passagers left outer join survivants on survivants.passengerid=passagers.passengerid group by sex,survived order by survived;

 * postgresql://postgres:***@localhost/postgres
4 rows affected.


[('female', 81, 0), ('male', 468, 0), ('female', 233, 1), ('male', 109, 1)]

In [73]:
%%sql

create view vivantmale as select Passengerid,name,sex,survived from survivants natural join passagers where survived=1 and sex='male' order by passengerid;
create view vivantfemale as select passengerid,name,sex,survived from survivants natural join passagers where survived=1 and sex='female'order by passengerid;

create view mortmale as select passengerid,name,sex,survived from survivants natural join passagers where survived=0 and sex='male' order by passengerid;
create view mortfemale as select passengerid,name,sex,survived from survivants natural join passagers where survived=0 and sex='female' order by passengerid;

 * postgresql://postgres:***@localhost/postgres
Done.
Done.
Done.
Done.


[]

In [74]:
%%sql
select * from vivantmale;
select * from vivantfemale;
select * from mortmale;
select * from mortfemale;

 * postgresql://postgres:***@localhost/postgres
109 rows affected.
233 rows affected.
468 rows affected.
81 rows affected.


[(15, 'Vestrom, Miss. Hulda Amanda Adolfina', 'female', 0),
 (19, 'Vander Planke, Mrs. Julius (Emelia Maria Vandemoortele)', 'female', 0),
 (25, 'Palsson, Miss. Torborg Danira', 'female', 0),
 (39, 'Vander Planke, Miss. Augusta Maria', 'female', 0),
 (41, 'Ahlin, Mrs. Johan (Johanna Persdotter Larsson)', 'female', 0),
 (42, 'Turpin, Mrs. William John Robert (Dorothy Ann Wonnacott)', 'female', 0),
 (50, 'Arnold-Franchi, Mrs. Josef (Josefine Franchi)', 'female', 0),
 (72, 'Goodwin, Miss. Lillian Amy', 'female', 0),
 (101, 'Petranec, Miss. Matilda', 'female', 0),
 (112, 'Zabour, Miss. Hileni', 'female', 0),
 (114, 'Jussila, Miss. Katriina', 'female', 0),
 (115, 'Attalah, Miss. Malake', 'female', 0),
 (120, 'Andersson, Miss. Ellis Anna Maria', 'female', 0),
 (133, 'Robins, Mrs. Alexander A (Grace Charity Laury)', 'female', 0),
 (141, 'Boulos, Mrs. Joseph (Sultana)', 'female', 0),
 (148, 'Ford, Miss. Robina Maggie "Ruby"', 'female', 0),
 (168, 'Skoog, Mrs. William (Anna Bernhardina Karlsson)', 'female', 0),
 (178, 'Isham, Miss. Ann Elizabeth', 'female', 0),
 (181, 'Sage, Miss. Constance Gladys', 'female', 0),
 (200, 'Yrois, Miss. Henriette ("Mrs Harbeck")', 'female', 0),
 (206, 'Strom, Miss. Telma Matilda', 'female', 0),
 (230, 'Lefebre, Miss. Mathilde', 'female', 0),
 (236, 'Harknett, Miss. Alice Phoebe', 'female', 0),
 (241, 'Zabour, Miss. Thamine', 'female', 0),
 (247, 'Lindahl, Miss. Agda Thorilda Viktoria', 'female', 0),
 (252, 'Strom, Mrs. Wilhelm (Elna Matilda Persson)', 'female', 0),
 (255, 'Rosblom, Mrs. Viktor (Helena Wilhelmina)', 'female', 0),
 (265, 'Henry, Miss. Delia', 'female', 0),
 (277, 'Lindblom, Miss. Augusta Charlotta', 'female', 0),
 (294, 'Haas, Miss. Aloisia', 'female', 0),
 (298, 'Allison, Miss. Helen Loraine', 'female', 0),
 (313, 'Lahtinen, Mrs. William (Anna Sylfven)', 'female', 0),
 (358, 'Funk, Miss. Annie Clemmer', 'female', 0),
 (363, 'Barbara, Mrs. (Catherine David)', 'female', 0),
 (375, 'Palsson, Miss. Stina Viola', 'female', 0),
 (397, 'Olsson, Miss. Elina', 'female', 0),
 (403, 'Jussila, Miss. Mari Aina', 'female', 0),
 (405, 'Oreskovic, Miss. Marija', 'female', 0),
 (410, 'Lefebre, Miss. Ida', 'female', 0),
 (416, 'Meek, Mrs. Thomas (Annie Louise Rowley)', 'female', 0),
 (420, 'Van Impe, Miss. Catharina', 'female', 0),
 (424, 'Danbom, Mrs. Ernst Gilbert (Anna Sigrid Maria Brogren)', 'female', 0),
 (437, 'Ford, Miss. Doolina Margaret "Daisy"', 'female', 0),
 (475, 'Strandberg, Miss. Ida Sofia', 'female', 0),
 (486, 'Lefebre, Miss. Jeannie', 'female', 0),
 (499, 'Allison, Mrs. Hudson J C (Bessie Waldo Daniels)', 'female', 0),
 (502, 'Canavan, Miss. Mary', 'female', 0),
 (503, "O'Sullivan, Miss. Bridget Mary", 'female', 0),
 (504, 'Laitinen, Miss. Kristina Sofia', 'female', 0),
 (535, 'Cacic, Miss. Marija', 'female', 0),
 (542, 'Andersson, Miss. Ingeborg Constanzia', 'female', 0),
 (543, 'Andersson, Miss. Sigrid Elisabeth', 'female', 0),
 (565, 'Meanwell, Miss. (Marion Ogden)', 'female', 0),
 (568, 'Palsson, Mrs. Nils (Alma Cornelia Berglund)', 'female', 0),
 (579, 'Caram, Mrs. Joseph (Maria Elias)', 'female', 0),
 (594, 'Bourke, Miss. Mary', 'female', 0),
 (611, 'Andersson, Mrs. Anders Johan (Alfrida Konstantia Brogren)', 'female', 0),
 (618, 'Lobb, Mrs. William Arthur (Cordelia K Stanlick)', 'female', 0),
 (635, 'Skoog, Miss. Mabel', 'female', 0),
 (639, 'Panula, Mrs. Juha (Maria Emilia Ojala)', 'female', 0),
 (643, 'Skoog, Miss. Margit Elizabeth', 'female', 0),
 (655, 'Hegarty, Miss. Hanora "Nora"', 'female', 0),
 (658, 'Bourke, Mrs. John (Catherine)', 'female', 0),
 (679, 'Goodwin, Mrs. Frederick (Augusta Tyler)', 'female', 0),
 (681, 'Peters, Miss. Katie', 'female', 0),
 (703, 'Barbara, Miss. Saiide', 'female', 0),
 (730, 'Ilmakangas, Miss. Pieta Sofia', 'female', 0),
 (737, 'Ford, Mrs. Edward (Margaret Ann Watson)', 'female', 0),
 (768, 'Mangan, Miss. Mary', 'female', 0),
 (773, 'Mack, Mrs. (Mary)', 'female', 0),
 (793, 'Sage, Miss. Stella Anna', 'female', 0),
 (800, 'Van Impe, Mrs. Jean Bap

Chez les hommes nous avons :

    109 hommes vivants,
    468 hommes morts.

Chez les femmes nous avons :

    233 femmes vivants,
    81 femmes morts.


In [52]:
import pandas
from sqlalchemy import create_engine
import plotly.express as px

engine = create_engine("postgresql://postgres:0000@localhost/postgres")
df1 = pandas.read_sql("select sex, count(survived),survived from passagers left outer join survivants on survivants.passengerid=passagers.passengerid group by sex,survived order by survived ;", engine)
fig = px.bar(df1, x="sex", y = "count", color = "survived")
fig.show()

<strong> Fin de SAE Merci!!</strong>